packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -0,0 +1,500 @@
+/// is a sheet route, than it will animate slightly upwards to look like it is laying
+/// on top of the previous stack of sheets.
+///
+/// Typically called by [showCupertinoSheet], which provides some boilerplate for
+/// pushing the `CupertinoSheetRoute` to the root navigator and providing simple
+/// nested naviagation.
+///
+/// The sheet will be dismissed by dragging downwards on the screen, or a call to
+/// [CupertinoSheetRoute.popSheet].
+///
+/// {@tool dartpad}
+/// This example shows how to navigate to [CupertinoSheetRoute] by using it the
+/// same as a regular route.
+///
+/// ** See code in examples/api/lib/cupertino/sheet/cupertino_sheet.0.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to setup a [CupertinoSheetRoute] with nested navigation.
+/// This allows for pushing to new routes that transition from within the sheet
+/// view.
+///
+/// This example can be acieved by calling [showCupertinoSheet], which sets up
+/// much of the same boilderplate by default.
+///
+/// ** See code in examples/api/lib/cupertino/sheet/cupertino_sheet.1.dart **
+/// {@end-tool}
+class CupertinoSheetRoute<T> extends PageRoute<T> with CupertinoSheetRouteTransitionMixin<T> {
+  /// Creates a page route that displays an iOS styled sheet.
+  CupertinoSheetRoute({
+    required this.builder,
+  });
+
+  /// Builds the primary contents of the sheet route.
+  final WidgetBuilder builder;
+
+  @override
+  DelegatedTransitionBuilder? get delegatedTransition => CupertinoSheetTransition.delegateTransition;
+
+  @override
+  Widget buildContent(BuildContext context) {
+    return _CupertinoSheetScope(
+      child: builder(context),
+    );
+  }
+
+  /// Checks if a Cupertino sheet view exists in the widget tree above the current
+  /// context.
+  static bool hasParentSheet(BuildContext context) {
+    return _CupertinoSheetScope.maybeOf(context) != null;
+  }
+
+  /// Pops the entire [CupertinoSheetRoute], if a sheet route exists in the stack.
+  ///
+  /// Used if to pop an entire sheet at once, if there is nested naviagtion within
+  /// that sheet.
+  static void popSheet(BuildContext context) {
+    if (hasParentSheet(context)) {
+      Navigator.of(context, rootNavigator: true).pop();
+    }
+  }
+
+  // Slightly darkens the sheet behind. Eyeballed from a simulator running iOS 18.0
+  // TODO(mitchgoodwin): Adjust with darkmode logic.
+  @override
+  Color? get barrierColor => const Color(0x20000000);
+
+  @override
+  bool get barrierDismissible => false;
+
+  @override
+  String? get barrierLabel => 'Stacked card appearance for modal bottom sheet';
+
+  @override
+  bool get maintainState => true;
+
+  @override
+  bool get opaque => false;
+}
+
+// Internally used to see if another sheet is in the tree already.
+class _CupertinoSheetScope extends InheritedWidget {
+  const _CupertinoSheetScope({
+    required super.child,
+  });
+
+  static _CupertinoSheetScope? maybeOf(BuildContext context) {
```

Comment:
Also all the call is side is not using this to build a different widget, this should use getInheritedWidgetOfExactType to not create a dependecy.
also updateShouldNotifty should just return false

Reply:
even if they do use this to build different content, i can think of a situation this will change in the life time of the widget in the subtree because it would only change if the route is disposed and the widget subtree would have been tear down already.